src/flowr/server-session.ts

```diff
@@ -0,0 +1,111 @@
+import * as net from 'net'
+import type * as vscode from 'vscode'
+import type { FlowrMessage } from '@eagleoutice/flowr/cli/repl'
+import type { FileAnalysisResponseMessageJson } from '@eagleoutice/flowr/cli/repl/server/messages/analysis'
+import type { SliceResponseMessage } from '@eagleoutice/flowr/cli/repl/server/messages/slice'
+import type { NodeId} from '@eagleoutice/flowr'
+import { visitAst } from '@eagleoutice/flowr'
+import type { SourceRange } from '@eagleoutice/flowr/util/range'
+import { isNotUndefined } from '@eagleoutice/flowr/util/assert'
+import { FlowrInternalSession } from './internal-session'
+
+export class FlowrServerSession {
```

Comment:
proposal: we should offer an abstract FlowrClient class as part of flowr which provides the basic logic like `connect(<address>, <port>)` which handles hello `request slicing` etc.